src/main/cpp/include/utils/alleles_combiner_template_definition.h

```diff
@@ -187,10 +230,53 @@ void AllelesCombiner::handle_allele(const size_t row_query_idx,  const STRING_VI
   //Track indexes of deletions and MNVs
   if(allele_config.is_deletion || allele_config.is_MNV)
     m_deletion_MNV_allele_idx_vec.push_back(allele_config.allele_idx);
+
+  //Update sample properties
+  sample_properties.contains_deletion = sample_properties.contains_deletion || allele_config.is_deletion;
+  sample_properties.contains_MNV = sample_properties.contains_MNV || allele_config.is_MNV;
+  sample_properties.num_deletions_or_MNVs += (allele_config.is_deletion || allele_config.is_MNV);
+  ++(sample_properties.num_alleles);
 }
 
-void AllelesCombiner::insert_allele_info(const size_t row_query_idx, const STRING_VIEW& REF,
-    const STRING_VIEW& delimited_ALT_str, const bool begins_before_curr_start_position) {
+template<typename ValidRowTrackerTy>
+void AllelesCombiner<ValidRowTrackerTy>::store_NON_REF_properties(SampleAlleleProperties& sample_properties,
+    AlleleConfig& allele_config) {
+  allele_config.is_symbolic_allele = true;
+  allele_config.is_NON_REF_allele = true;
+  sample_properties.NON_REF_allele_idx = allele_config.allele_idx;
+}
+
+template<typename ValidRowTrackerTy>
+bool AllelesCombiner<ValidRowTrackerTy>::process_allele_char(const char curr_char, SampleAlleleProperties& sample_properties,
```

Comment:
Would it be better to use a regular expression to parse the entire delimited allele string?

Reply:
Could you elaborate?

Reply:
I meant could you use regular expressions with groups to capture strings say between < and > or between [ and ], etc. The delimited allele string is not easily parseable I guess.

Reply:
The string processing is not very complex - regular expressions would be an overkill. A simple single pass over the string is sufficient for now.